crowbar_framework/spec/controllers/api/restart_management_controller_spec.rb

```diff
@@ -0,0 +1,209 @@
+# limitations under the License.
+#
+
+require "spec_helper"
+require "json"
+
+describe Api::RestartManagementController, type: :request do
+  let(:headers) { { ACCEPT: "application/vnd.crowbar.v2.0+json" } }
+
+  before(:each) do
+    @catalog = {
+      "neutron" => 10,
+      "nova" => 20,
+      "cinder" => 30
+    }
+    allow_any_instance_of(BarclampCatalog).to receive(:members).and_return(@catalog)
+    @nova_proposal = Proposal.where(barclamp: "nova", name: "default").create(
+      barclamp: "nova",
+      name: "default"
+    )
+    @cinder_proposal = Proposal.where(barclamp: "cinder", name: "default").create(
+      barclamp: "cinder",
+      name: "default"
+    )
+    @testing_node = NodeObject.find_by_name("testing.crowbar.com")
+    @admin_node = NodeObject.find_by_name("admin.crowbar.com")
+    @nova_proposal.raw_data["deployment"]["nova"]["elements"] = {
+      "openstack-nova-api" => ["testing.crowbar.com"]
+    }
+
+    @cinder_proposal.raw_data["deployment"]["cinder"]["elements"] = {
+      "openstack-cinder-api" => ["testing.crowbar.com"]
+    }
+    allow(Proposal).to receive(:where).with(barclamp: "neutron").and_return([])
+    allow(Proposal).to receive(:where).with(barclamp: "nova").and_return([@nova_proposal])
+    allow(Proposal).to receive(:where).with(barclamp: "cinder").and_return([@cinder_proposal])
+    allow(NodeObject).to receive(:find_by_name).with("testing.crowbar.com").and_return(
+      @testing_node
+    )
+  end
+
+  context "GET sync" do
+    it "lists the barclamps and their disallowed_restart status" do
+      # set only nova to disallow restart
+      @testing_node.set["nova"]["disallow_restart"] = true
+
+      get "/api/restart_management/sync", {}, headers
+      expect(response).to have_http_status(:ok)
+      expect(ActiveSupport::JSON.decode(response.body)).to eq("nova" => true, "cinder" => false)
+    end
+  end
+
+  context "POST sync" do
+    it "enabling changes the barclamp disallow_restart status to true" do
+      # disable_restart should not exists prior to the call
+      expect(@testing_node["nova"][:disallow_restart]).to be(nil)
+
+      post "/api/restart_management/sync", { disallow_restart: true, barclamp: "nova" }, headers
+      expect(response).to have_http_status(:ok)
+      # now the flag should be on true
+      expect(@testing_node["nova"][:disallow_restart]).to be(true)
+    end
+
+    it "disabling changes the barclamp disallow_restart status to false" do
+      # disable_restart should not exists prior to the call
+      expect(@testing_node["nova"][:disallow_restart]).to be(nil)
+
+      post "/api/restart_management/sync", { disallow_restart: false, barclamp: "nova" }, headers
+      expect(response).to have_http_status(:ok)
+      # now the flag should be on false
+      expect(@testing_node["nova"][:disallow_restart]).to be(false)
+    end
+
+    it "alternate calls to enable/disable set the barclamp disallow restart status properly" do
```

Comment:
I don't get why do we need this test. We already tested the status what happens when we set it to `true` then what happens if we set it to `false`. What are we testing by setting it `true`, `false`, `true` ? Also the comments are misleading, as they are out of sync.

Reply:
well, we are only testing the transition from a know mocked state into a desired state with the other tests, so I think its useful to test the toggling of the value to check that it works as expected when the mocked state has gone away and we only have real values in there. Plus it doesnt hurt to have more tests.